app/common/tdebounce/tdebounce.directive.js

```diff
@@ -0,0 +1,38 @@
+(function() {
+	'use strict';
+
+	angular
+		.module('tdebounce')
+	  	.directive('tDebounce', tDebounce);
+
+	/* @ngInject */
+	function tDebounce() {
+		var directive = {
+
+			link: link,
+			restrict: 'A',
+			scope: {
+				tDebounce: '&',
+				tDebounceDelay: '='
+			}
+		};
+
+		return directive;
+
+		function link(scope, element, attrs){
+			var initial;
+
+			element.bind('keydown', function() {
+				clearTimeout(initial);
+				startTimeout();
+			});
+
+			function startTimeout() {
+				initial = window.setTimeout(function() {
```

Comment:
You should inject $window use this instead, i think it could save you the scope.$apply.